pkg/server/server.go

```diff
@@ -89,7 +89,10 @@ func (host *yamlLanguageHost) GetRequiredPlugins(ctx context.Context,
 	pkgs, pluginDiags := pulumiyaml.GetReferencedPlugins(template)
 	diags.Extend(pluginDiags...)
 	if diags.HasErrors() {
-		return nil, diags
+		// We currently swallow the error to allow project config to evaluate
+		// Specifically, if one sets a config key via the CLI but not within the `config` block
```

Comment:
This seems odd? Why does yaml program parsing require stack config at all?

Reply:
It doesn't require stack config specifically, the issue is we don't have a config node for project config until runtime. So when it's parsing and sees something like `${myConfigKey}`, where we don't have `myConfigKey` under the `config` block, it would treat it as an unknown resource and error

Reply:
Determining referenced plugins requires analyzing what providers are created, if they are set to default, if resources specify explicit providers, ect. This is all done after we perform the top-sort on nodes. Missing config nodes are caught by top-sort and we error out. This error would prevent the rest of the `pulumi up`, even when the update itself is valid.

Reply:
This sounds like a TODO to fix, is there an issue for it?

Reply:
Oh I see theres already https://github.com/pulumi/pulumi-yaml/pull/392 on it